File: server/DocumentsKM/Models/BoltDiameter.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace DocumentsKM.Models
{
    // Диаметр болта
    public class BoltDiameter
    {
        [Key]
        public Int16 Id { get; set; }

        // Диаметр
        [Required]
        public Int16 Diameter { get; set; }

        // Вес гайки
        [Required]
        public float NutWeight { get; set; }

        // Сталь шайбы
        [Required]
        [MaxLength(50)]
        public string WasherSteel { get; set; }

        // Вес шайбы
        [Required]
        public float WasherWeight { get; set; }

        // Толщина шайбы
        [Required]
        [Range(0, 65535, ErrorMessage = "Value should be greater than or equal to 0")]
        public Int16 WasherThickness { get; set; }

        // Нормативный документ на болты
        [Required]
        [MaxLength(50)]
        public string BoltTechSpec { get; set; }

        // Класс прочности болтов
        [Required]
        [MaxLength(50)]
        public string StrengthClass { get; set; }

        // Нормативный документ на гайки
        [Required]
        [MaxLength(50)]
        public string NutTechSpec { get; set; }

        // Нормативный документ на шайбы
        [Required]
        [MaxLength(50)]
        public string WasherTechSpec { get; set; }
    }
}

File: client/src/model/BoltDiameter.ts
interface IBoltDiameter {
	id: number
	diameter: number
	nutWeight: number
	washerSteel: string
	washerWeight: number
	washerThickness: number
	boltTechSpec: string
	strengthClass: string
	nutTechSpec: string
	washerTechSpec: string
}

class BoltDiameter {
	id: number
	diameter: number
	nutWeight: number
	washerSteel: string
	washerWeight: number
	washerThickness: number
	boltTechSpec: string
	strengthClass: string
	nutTechSpec: string
	washerTechSpec: string

	constructor(obj?: IBoltDiameter) {
		this.id = (obj && obj.id) || 0
		this.diameter = (obj && obj.diameter) || 0
		this.nutWeight = (obj && obj.nutWeight) || 0
		this.washerSteel = (obj && obj.washerSteel) || ''
		this.washerWeight = (obj && obj.washerWeight) || 0
		this.washerThickness = (obj && obj.washerThickness) || 0
		this.boltTechSpec = (obj && obj.boltTechSpec) || ''
		this.strengthClass = (obj && obj.strengthClass) || ''
		this.nutTechSpec = (obj && obj.nutTechSpec) || ''
		this.washerTechSpec = (obj && obj.washerTechSpec) || ''
	}
}

export default BoltDiameter
